### PR TITLE
Feature/archive projects

### PR DESCRIPTION
**Pull request by Kucziii**
[add] Added archive method in projects controller.
[add] Added "archived" attribute to project migration.
[add] Added tests for archive method in RSpec
[add] Added API method for checking the current user in react component.
[add] Added react component for archivising projects by admin (modal + button styling)
[modify] Modified project-list component to communicate with API and display archive button only if user is admin

I added archive method for admin to make him able to archivise old projects and leave them in database without deleting them. RSpec test confirms that it's working but integration with new functionality from teammates branches is needed to confirm it. If the current user is admin, in the project list appears additional button for archivising. Confirming modal appears after clicking.

API is used to inform react component whether current user is admin or not. It's important because standard user shouldn't be able to archivise projects.

Didn't add styling for archivised projects on the list as merging all current branches is needed for that.

### CODE REVIEW

File: app/javascript/packs/projects-list.jsx [hunk 1]
```diff
@@ -12,20 +14,43 @@ export default class ProjectList extends React.PureComponent {
       .then(projects => {
         this.setState({ projects: projects.data });
       });
+
+    axios
+      .get("http://localhost:3000/api/v1/user.json")
+      .then(response => {
+        console.log(response);
```
Comment (Vakrim) [inline app/javascript/packs/projects-list.jsx]: remove console logs used for debugging

File: app/javascript/packs/archive_modal.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false
+    };
+
+    this.toggle = this.toggle.bind(this);
+  }
+
+  toggle() {
```
Comment (Vakrim) [inline app/javascript/packs/archive_modal.jsx]:
```suggestion
  toggle = () => {
```
you can use arrow function to skip binding function in constructor

File: app/controllers/api/v1/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Api::V1::UsersController < ApplicationController
+  def index
+    @user = current_user.admin
+    render json: @user
```
Comment (hirokiraj) [inline app/controllers/api/v1/users_controller.rb]: prefer using jbuilder

File: app/controllers/project_controller.rb [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
-  before_action :set_project, only: %i[show edit update destroy]
+  before_action :set_project, only: %i[show edit update destroy archive]
 
   def index
     @projects = Project.all
   end
 
-  def show; end
+  def show
+  end
```
Comment (hirokiraj) [inline app/controllers/project_controller.rb]: is this action empty? If yes then let's just remove it

File: app/controllers/project_controller.rb [hunk 5]
```diff
@@ -32,6 +33,11 @@ def destroy
     head :no_content
   end
 
+  def archive
+    @project.update_attribute(:archived, true)
```
Comment (hirokiraj) [inline app/controllers/project_controller.rb]: just `update`

File: app/controllers/project_controller.rb [hunk 6]
```diff
@@ -32,6 +33,11 @@ def destroy
...
+    head :no_content
```
Comment (hirokiraj) [inline app/controllers/project_controller.rb]: why?

File: config/routes.rb [hunk 7]
```diff
@@ -7,9 +7,15 @@
...
 
-
   resources :manage_io, only: [:index]
 
   root to: 'manage_io#index'
+  get '/project/archive/:id' => 'project#archive', :defaults => { :format => :json }
+  
+  namespace :api do
+    namespace :v1 do
+        resources :users
```
Comment (hirokiraj) [inline config/routes.rb]: are all of them paths required?

File: coverage/.last_run.json [hunk 8]
```diff
@@ -1,5 +1,5 @@
 {
```
Comment (hirokiraj) [inline coverage/.last_run.json]: remove coverage report

File: spec/controllers/project_controller_spec.rb [hunk 9]
```diff
@@ -85,4 +85,11 @@
       expect { subject }.to change(Project, :count).by(-1)
     end
   end
+
+  describe '#archive' do
+    let(:project) { create(:project) }
+    before { get :archive, params: { id: project.id } }
+
+    it { expect(Project.find(project.id).archived).to eq(true) }
```
Comment (hirokiraj) [inline spec/controllers/project_controller_spec.rb]: it is already fetched under `project` variable, you just need to reload it

File: app/controllers/project_controller.rb [hunk 10]
```diff
@@ -1,13 +1,14 @@
 # frozen_string_literal: true
 
 class ProjectController < ApplicationController
```
Comment (hirokiraj) [inline app/controllers/project_controller.rb]: shouldn't it be in api namespace

**Pull request merged**